src/java/org/apache/cassandra/index/sai/disk/StorageAttachedIndexWriter.java

```diff
@@ -303,7 +303,7 @@ public void abort(Throwable accumulator, boolean fromIndex)
         
         // For non-compaction, make any indexes involved in this transaction non-queryable, as they will likely not match the backing table.
         // For compaction: the compaction task should be aborted and new sstables will not be added to tracker
-        if (fromIndex && opType != OperationType.COMPACTION)
```

Comment:
So this change means if we abort during a flush we should just abort the index creation and not fail the index?  Please update the comment above as well.

Reply:
I updated the comment and added more checks to the test to verify the sstable is not created on index build failure.